code/datums/components/quality.dm

```diff
@@ -72,14 +77,13 @@
 	parent_item.armor?.modifyAllRatings(armor_qual) // modifies all armor ratings
 	if(quality_level == 10)
 		parent_item.AddComponent(/datum/component/fantasy,10)
-	parent_item.name = handle_name(old_parent_item.name)
+	parent_item.name = handle_name(parent_item.name)
 
 ///Returns the item to the state it was before the modification
 /datum/component/quality/proc/unmodify()
 	var/obj/item/parent_item = parent
 	var/quality = quality_levels[quality_level+1] //lists start with 1
-	parent_item = old_parent_item
-	parent_item.name = old_parent_item.name
+	parent_item.name = initial(parent_item.name)
```

Comment:
This will fuck up if you do it on a warhammer or something because those get a name prefixed from materials